### PR TITLE
fix wrong handling of M600 and emulated M600

### PR DESCRIPTION
**Pull request by guruathwal**
- Fix wrong handling of M600
- Fix wrong handling of emulated M600
- Stop nozzle park commands on pause through non-emulated M600 causing the buffer to fill up while the printer and UI freeze.

### CODE REVIEW

File: TFT/src/User/API/interfaceCmd.c [hunk 1]
```diff
@@ -247,26 +233,21 @@ void sendQueueCmd(void)
...
-            dequeueLastCmd(gcode);
-
-            if (printPause(true, true))  // if print successfully paused, send the gcode
-              sendDequeuedCmd(gcode, avoid_terminal);
-            return;
+            // pasue if printing form TFT and purge M0/M1 command.
+            if (infoFile.source < BOARD_SD )
+            {
+              purgeLastCmd(avoid_terminal);
+              if (!isPaused()) printPause(true, PAUSE_M0);
```
Comment (digant73) [inline TFT/src/User/API/interfaceCmd.c:240]: I would remove "if (!isPaused())", the check is already done in printPause().
I would also put a blank line before printPause (as done in other invokations in interfaceCmd) just to highlight that printPause is an external function

File: TFT/src/User/API/interfaceCmd.c [hunk 2]
```diff
@@ -359,7 +344,8 @@ void sendQueueCmd(void)
                   Serial_Puts(SERIAL_PORT_2, "\n");
                 }
                 Serial_Puts(SERIAL_PORT_2, "ok\n");
-                purgeLastCmd();
+                purgeLastCmd(avoid_terminal);
+
```
Comment (digant73) [inline TFT/src/User/API/interfaceCmd.c:348]: no blank line was used for "return" in interfaceCmd. So it should be removed here

File: TFT/src/User/API/interfaceCmd.c [hunk 3]
```diff
@@ -912,27 +895,31 @@ void sendQueueCmd(void)
...
-            purgeLastCmd();
-
-            printPause(true, false);
-            return;
+            // purge and pause only if emulated M600 is enabled.
+            // if emulated M600 is disabled then let the printer pause the print to avoid premature pause
+            if (infoSettings.emulate_m600 == 1)
+            {
+              purgeLastCmd(avoid_terminal);
+              printPause(true, PAUSE_M600);
```
Comment (digant73) [inline TFT/src/User/API/interfaceCmd.c:905]: as reported in my first comment, I would put a blank line before printPause()

File: TFT/src/User/API/interfaceCmd.c [hunk 4]
```diff
@@ -912,27 +895,31 @@ void sendQueueCmd(void)
...
-              purgeLastCmd();
-
-              printPause(true, false);
-              return;
+              // purge and pause only if emulated M600 is enabled.
+              // if emulated M600 is disabled then let the printer pause the print to avoid premature pause
+              if (infoSettings.emulate_m600 == 1)
+              {
+                purgeLastCmd(avoid_terminal);
+                printPause(true, PAUSE_M600);
```
Comment (digant73) [inline TFT/src/User/API/interfaceCmd.c:920]: as reported in my first comment, I would put a blank line before printPause()

**Review by digant73 — CHANGES_REQUESTED** [review]

File: TFT/src/User/API/interfaceCmd.c (hunk 1 above)
Comment (guruathwal) [reply TFT/src/User/API/interfaceCmd.c:240]: To see the source of the function I would right-click on it and select `Goto to Definition`.
Frankly speaking, I am very much confused with the rules you prefer to put line spacing between code line. How do you identify what the line spacing was for? it just ends up with spacing on every line. I prefer simple spacing rule. Line spacing between variable assignments and function calls.

Comment (digant73) [reply TFT/src/User/API/interfaceCmd.c:240]: different style. In any case, if you use your style apply it to the rest of the file. So remove the blank lines where still present

File: TFT/src/User/API/Printing.c [hunk 5]
```diff
@@ -456,64 +457,70 @@ bool printPause(bool is_pause, bool is_m0pause)
...
-                       infoSettings.pause_feedrate[FEEDRATE_XY]);
+          coordinateGetAll(&tmp);
+
+          if (isCoorRelative == true)    mustStoreCmd("G90\n");
+          if (isExtrudeRelative == true) mustStoreCmd("M82\n");
+
+          if (heatGetCurrentTemp(heatGetCurrentHotend()) > infoSettings.min_ext_temp)
+          {
+            mustStoreCmd("G1 E%.5f F%d\n", tmp.axis[E_AXIS] - infoSettings.pause_retract_len,
+                        infoSettings.pause_feedrate[FEEDRATE_E]);
```
Comment (digant73) [inline TFT/src/User/API/Printing.c:476]: one more space to align the code

File: TFT/src/User/API/Printing.c [hunk 6]
```diff
@@ -456,64 +457,70 @@ bool printPause(bool is_pause, bool is_m0pause)
...
+          }
+
+          if (coordinateIsKnown())
+          {
+            mustStoreCmd("G1 Z%.3f F%d\n", tmp.axis[Z_AXIS] + infoSettings.pause_z_raise, infoSettings.pause_feedrate[FEEDRATE_Z]);
+            mustStoreCmd("G1 X%.3f Y%.3f F%d\n", infoSettings.pause_pos[X_AXIS], infoSettings.pause_pos[Y_AXIS],
+                        infoSettings.pause_feedrate[FEEDRATE_XY]);
```
Comment (digant73) [inline TFT/src/User/API/Printing.c:483]: one more space to align the code

**Review by digant73 — CHANGES_REQUESTED** [review]

**Pull request merged**